lib/ecco/row_event.rb

```diff
@@ -1,3 +1,34 @@
 module Ecco
-  RowEvent = Struct.new(:type, :table_id, :database, :table, :rows)
+  class RowEvent
+    attr_accessor :type, :table_id, :database, :table, :rows
+
+    def initialize
+      @table_id = nil
+      @database = nil
+      @table = nil
+      @rows = nil
+      @type = nil
+    end
+  end
+
+  class RowEventWrite < RowEvent
+    def initialize
+      super
+      @type = "WRITE_ROWS"
```

Comment:
Just scanning this file from top down and this caught my attention, we have a class name that implies write row, but yet there is an instance variable saying the same thing. Something is redundant :)

Reply:
Perhaps back out on this change (i.e. use the plain old struct we had before) and focus on just getting `#rows` right, just the "mapping-over"-part

Reply:
It might be possible to remove `@type` altogether, I think Bruce uses it to conditionalize aspects of its code.
From Bruce `event.rb` line 74:

```
def self.extract_event_type(row_event)
    case row_event.type
    when WRITE_ROW_EVENT_TYPE_REGEXP
      "insert"
    when UPDATE_ROW_EVENT_TYPE_REGEXP
      "update"
    when DELETE_ROW_EVENT_TYPE_REGEXP
      "delete"
    end
  end
```

This could maybe be moved into the classes instead. Gonna continue digging in Bruce and see if `type` is used elesewhere.